corehq/apps/sms/api.py

```diff
@@ -182,6 +182,12 @@ def send_message_via_backend(msg, backend=None, orig_phone_number=None):
     except Exception:
         logging.exception("Could not clean text for sms dated '%s' in domain '%s'" % (msg.date, msg.domain))
     try:
+        if not domain_has_privilege(msg.domain, privileges.OUTBOUND_SMS):
+            raise Exception(
```

Comment:
Would be nice if we used a separate Exception subclass (something like MissingPrivilegeException), but I don't feel too strong about that.